raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -830,3 +835,213 @@ def run_test_automatic_dispute(raiden_network, deposit, token_addresses):
+        chain_state=views.state_from_app(alice_app),
+        token_network_id=token_network_identifier,
+    )
+
+    channel_identifier = get_channelstate(alice_app, bob_app, token_network_identifier).identifier
+
+    assert channel_identifier in token_network.partneraddresses_to_channelidentifiers[
+        bob_app.raiden.address
+    ]
+
+    alice_to_bob_amount = 10
+    identifier = 1
+
+    alice_transfer_secret = sha3(alice_app.raiden.address)
+    alice_transfer_secrethash = sha3(alice_transfer_secret)
+
+    bob_transfer_secret = sha3(bob_app.raiden.address)
+    bob_transfer_secrethash = sha3(bob_transfer_secret)
+
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=alice_transfer_secrethash,
+    )
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=bob_transfer_secrethash,
+    )
+
+    alice_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=bob_app.raiden.address,
+        identifier=identifier,
+        secret=alice_transfer_secret,
+    )
+
+    bob_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=alice_app.raiden.address,
+        identifier=identifier + 1,
+        secret=bob_transfer_secret,
+    )
+
+    wait_for_state_change(
+        raiden=alice_app.raiden,
+        item_type=ActionInitTarget,
+        attributes={},
+        retry_timeout=retry_timeout,
+    )
+
+    wait_for_state_change(
+        raiden=bob_app.raiden,
+        item_type=ActionInitTarget,
+        attributes={},
+        retry_timeout=retry_timeout,
+    )
+
+    alice_bob_channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    alice_lock = channel.get_lock(alice_bob_channel_state.our_state, alice_transfer_secrethash)
+    bob_lock = channel.get_lock(alice_bob_channel_state.partner_state, bob_transfer_secrethash)
+
+    # This is the current state of protocol:
+    #
+    #    A -> B LockedTransfer
+    #    B -> A SecretRequest
+    #    - protocol didn't continue
+    assert_synced_channel_state(
+        token_network_identifier=token_network_identifier,
+        app0=alice_app,
+        balance0=deposit,
+        pending_locks0=[alice_lock],
+        app1=bob_app,
+        balance1=deposit,
+        pending_locks1=[bob_lock],
+    )
+
+    # A ChannelClose event will be generated, this will be polled by both apps
+    # and each must start a task for calling settle
+    RaidenAPI(bob_app.raiden).channel_close(
+        registry_address=registry_address,
+        token_address=token_address,
+        partner_address=alice_app.raiden.address,
+    )
+
+    secret_registry_proxy = alice_app.raiden.chain.secret_registry(
+        secret_registry_address,
+    )
+    secret_registry_proxy.register_secret(
```

Comment:
This works but it's unnecessary, we only need to have an unlock that benefits our channel's partner, expired locks work fine for this.